### ISSUE TITLE
Warn on non-shell comments.

### ISSUE THREAD

**Issue by arth1** [koalaman/shellcheck#865]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

// This is not a comment
/* This is not a comment either
# although this is */

```

#### Here's what I wanted or expected to see:

In test.sh line 3:
// This is not a comment
^-- SC7777:  Root inodes cannot be executed for any known file system.

In test.sh line 4:
/* This is not a comment either
^-- SC7778: A * glob on root can never return a single executable

#### Comment

Yes, I have seen this.

**Comment by Vdragon** [comment]
IMO user should know the comment syntax is wrong once they saw ShellCheck spills errors on their comments.

**Comment by arth1** [comment]
@Vdragon : You miss the point:  shellcheck does not report _any_ errors.
From shellcheck's point of view, the code is fine.

// and /* start with a /, so are presumed to be the path to a command, and the rest of the line arguments.  That validates without warnings.

**Comment by Vdragon** [comment]
I misunderstood your issue, I'm fine by then.

**Comment by arth1** [comment]
After further consideration, the /* should probably not be warned against.  Depending on your file system, you can have a file named '*' in the root directory:

$ /* this is a comment */
Are you trying to execute a comment?
$ cat /\*
#!/bin/sh
echo "Are you trying to execute a comment?"

However, I can see no possible way that // can be a legal command, and I have seen // used in bash scripts which worked fine until the execution path that had // in it failed.

**Comment by koalaman** [comment]
I managed to find one single example of this. The majority of cases were php and C code that somehow had made it onto a bash IRC channel.

Given the number of thumbs-ups and all other metrics being neutral, I've added it anyways. It's available in a6fb9d1

It also does warn about `/*`. If someone wants a file with that name, that's up to them, but they need to quote it correctly.
